src/main/kotlin/Main.kt

```diff
@@ -28,18 +19,20 @@ object Main {
         // Treepat Parsing
         val treepatExpression = TreepatExpression.createFromFile(args[0])
         // Tree File Parsing
-        val targetTree =
-            DefaultTargetTree<DefaultTargetTreeNode>(args[1])
-
-        val functionResult = targetTree.findMatchesRaw(treepatExpression)
-
-        val solutions: List<String>
-        solutions = if (functionResult.hasMatch) {
-            functionResult.responses.map { targetTree.root?.matchedNodesString(it.matches)?.trimIndent() ?: "" }
+        val targetTree = DefaultTargetTree.createFromFile(args[1])
+        // Match Manager
+        val manager = MatchManager(treepatExpression, targetTree)
+        var match = manager.getNextMatch()
+        if (match == null) {
+            print("Match not found")
         } else {
-            listOf("Match not found")
+            print(manager.matchToString(match))
+            match = manager.getNextMatch()
+            while (match != null) {
+                print("\n-\n" + manager.matchToString(match))
+                match = manager.getNextMatch()
+            }
```

Comment:
Now, this is how I printed the matches as they were found in the `Main` using `getNextMatch()`.

What do you think about it?